primary/src/primary.rs

```diff
@@ -487,13 +491,20 @@ impl<PublicKey: VerifyingKey> PrimaryToPrimary for PrimaryReceiverHandler<Public
                     from: from.clone(),
                 })
                 .await
-                .expect("Failed to send primary message"),
+                .map_err(|_| DagError::ShuttingDown),
+
+            PrimaryMessage::Reconfigure(notification) => self
+                .tx_state_handler
+                .send(notification)
+                .await
+                .map_err(|_| DagError::ShuttingDown),
             _ => self
                 .tx_primary_messages
                 .send(message)
                 .await
-                .expect("Failed to send certificate"),
+                .map_err(|_| DagError::ShuttingDown),
```

Comment:
you are completely right. I will move this message to the primary-worker channel (that is not exposed to WAN) until a next PR does what Brandon suggested

Reply:
The suggestion I had was to have a new gRPC service that only handles these types of messages. Multiple gRPC services can hang off of the same gRPC server so we wouldn't need to use a whole new interface if we didn't want to.

Reply:
> you are completely right. I will move this message to the primary-worker channel (that is not exposed to WAN) until a next PR does what Brandon suggested

I am ok with this as a trade off to make this PR progress but I would like to ask to prioritise the next PR to address this properly.

@bmwill If we use a new gRPC service and attached it under the same gRPC server then this will still be exposed under the same port, right? If that's true then it's not clear to me how we could prevent other primaries from accessing the endpoints of this service (given that we don't have any kind of authentication - yet) . Or did you have in mind some other way?